### ISSUE TITLE
Cannot catch a certain crash with try/catch or errorhandler

### ISSUE THREAD

**Issue by UserSina** [breejs/bree#88]
### **TL;DR**:  _The final piece of code (scroll to bottom) is crashing my express server, I want to catch the error instead._

I have a nodejs/express environment that has some routes for front end. It also has some running workers with a breejs instance.
The instance executes a query to an  Oracle database and logs the results every period of time.

I get an Oracle error when the job runs for a certain number of times (15 times or so) which I can't catch and my express app **always** crashes, the error is this:
`ORA-24550: signal received: Unhandled exception: Code=c0000005 Flags=0`

**_I simply want to find a solution to catch the error instead of crashing the server (try/catch has no effect)._**

Below is my full code (or the gist of it) from top to bottom:

`app.js: `
```JAVASCRIPT
// ... Express & routes imports
// require scheduler
const Scheduler = require('./Scheduler');
var scheduler = Scheduler;

var app = express();

// Start the scheduler after 5 seconds
setTimeout(async () => {
  await scheduler.startScheduler();
}, 5000);

// setup routes
app.use('/', indexRouter);
app.use('/users', usersRouter);

// catch 404 and forward to error handler
app.use(function (_req, _res, next) {
  next(createError(404));
});

// error handler (*doesn't catch the error related to the database in worker)
app.use(function (err, _req, res, _next) {
  res.status(err.status || 500);
  res.json({
    status: 'fail',
    code: 1,
    message: err.message,
  });
});

module.exports = app;
```

`Scheduler.js: `
```JAVASCRIPT
const Bree = require('bree');
const logger = require('../location/to/logger');

class Scheduler {
  constructor() {
    this.bree = null;
  }

  // - Create a new scheduler and assign it to bree
  startScheduler = async () => {
    var queries = await getMyQueries(); // Example: [ { "id": 1, "body": "SELECT * FROM my_table" }, ... ]
    this.bree = this.scheduleJobs(queries);
    this.bree.start();
  };
  
  // - Create the jobs from a list of objects {}
  scheduleJobs = (list) => {
    let myJobs = [];
    list.forEach((query) => {
      myJobs.push({
        name: `worker-${query.id}`,
        path: path.join(__dirname + '/jobs', 'my-worker.js'), // File below
        interval: 10000,
        worker: {
          workerData: query,
        },
      });
    });
	
    return new Bree({
      root: false,
      logger,
      jobs: myJobs,
      errorHandler: (error, workerMetaData) => {
        // Does not catch: ORA-24550 (I get no logs & my server crashes)
        logger.error(error);
        logger.warn(
          `There was an error while running a worker: ${workerMetaData.name}`
        );
      },
    });
  };
}

module.exports = new Scheduler();
```

The main worker file `my-worker.js: `
```JAVASCRIPT
const { workerData, parentPort } = require('worker_threads');
const { isValid, executeOracleQuery } = require('../path/to/file');

const terminate = () => {
  if (parentPort) parentPort.postMessage('done');
  else process.exit(0);
};

// Execute on entry
(async () => {
  var myQuery = workerData.job.worker.workerData;
  
  // Exist if permission denied
  if (!isValid()) {
    logger.error('Permission denied, terminating worker!', myConfig);
    terminate();
  }  

  // Execute the query in oracle database
  var result = await executeOracleQuery(myQuery);
  result.status === 'fail'
    ? logger.error('Could not save statistic', result)
    : logger.info('Statistic saved successfully');
  
  terminate();
})();
```
And finally the `executeOracleQuery` function which is likely causing the crash:
```JAVASCRIPT
const executeOracleQuery = async (sqlQuery) => {
  let params = { host: 'my', port: 'connection', database: 'parameters' }
  
  // Create connection
  try {
    const oracledb = require('oracledb');
    const conn = await oracledb.getConnection({
        user: params.user,
        password: params.password,
        connectString: `${params.host}:${params.port}/${params.database}`
    })
  } catch (error) {
	return { status: 'fail', error };
  }
  
  // Execute my query
  try {
    let data = await conn.execute(sqlQuery);
    await conn.close();
    return { status: 'success', data };
  } catch (error) {
    return { status: 'fail', error };
  }
}
```
I do not need to fix the ORA error, I simply want to prevent my app from crashing atm.

**Comment by naz** [comment]
I've seen funky handling happening when using native `worker_threads` instead of recommended by bree - [bthreads](https://github.com/chjj/bthreads) library. Try swapping them out in your `my-worker.js` module.

**Comment by UserSina** [comment]
Using `bthreads` instead of `worker_threads` does indeed prevent the app from crashing when the worker is running.
However when the worker is online (still running) and I try to go to a route in my front (GET for example), the server simply crashes.

Note that this does not happen when opening a connection with `mysql2` for example, only happens -weirdly enough- for `oracledb`.

**Comment by shadowgate15** [comment]
In your worker, will your logger `throw` an error? If the worker doesn't `throw` the error, then `errorHandler` won't be called.

**Comment by UserSina** [comment]
I don't throw the error directly (since I handle them separately) however the `oracledb` should be the one that throws the error since my server crashes when an instance is active.

I've tried promisifying the `executeOracleQuery()` function but my server still crashes.

**Comment by UserSina** [comment]
Tried starting with `node` and now I get:

```JAVASCRIPT
npm ERR! code ELIFECYCLE
npm ERR! errno 3221225477
```
Check [full error log](https://github.com/oracle/node-oracledb/issues/1342#issuecomment-808140284).

**Comment by shadowgate15** [comment]
fixed by `oracledb`
https://github.com/oracle/node-oracledb/issues/1342
